### PR TITLE
Add building if satellites are nearby

### PR DESCRIPTION
**Pull request by rzulian**
This solves #113 
I've added bug's test, with the latest mine in a different position.
The only problem is that is exceding the timeout and I'm not able to solve it

### CODE REVIEW

File: src/player.ts [hunk 1]
```diff
@@ -315,13 +315,17 @@ export default class Player extends EventEmitter {
     // Add to nearby federation
     if (building !== Building.GaiaFormer && !hex.belongsToFederationOf(this.player)) {
       const group: GaiaHex[] = this.buildingGroup(hex);
+      const sats: GaiaHex[] = this.satelittesNearby(map, hex);
       const hasFederation = group.some(hx => hx.belongsToFederationOf(this.player));
```
Comment (coyotte508) [inline src/player.ts]: The only change needed is here: 

```
const hasFederation = group.concat(map.grid.neighbours(hex)).some(hx => hx.belongsToFederationOf(this.player));
```

And that's it. No other line necessary. Just this one line :)

With the current code, a new mine next to a satellite will get added to the federation, but any other building next to it (but far away from the satellite) won't.

Comment (rzulian) [reply src/player.ts]: oh yes, much cleaner.
BTW, isn't the current code doing what you are saying anyway, given that the group code is the same?

Comment (coyotte508) [reply src/player.ts]: The problem is that you don't set `hasFederation` to `true` if a satellite has the federation. 

So this:

```
      if (hasFederation) {
        // Add federation to ALL buildings of the group, not just the new buidling
        for (const h of group) {
          h.addToFederationOf(this.player);
        }
      }
```

is not triggered!

**Comment by rzulian** [comment]
I've tried to use function with time out, but I was getting `non-arrow functions are forbidden (only-arrow-functions)`, even though you are the same in a previous test with error

**Comment by coyotte508** [comment]
You shouldn't get the error if you have `this` in the body of your function.

**Pull request merged**